samples/bluetooth/mesh/sensor_client/src/model_handler.c

```diff
@@ -36,56 +33,142 @@ static void sensor_cli_data_cb(struct bt_mesh_sensor_cli *cli,
+{
+	printk("Descriptor of sensor with ID 0x%04x:\n", sensor->id);
+	printk("\ttolerance: { positive: %s, negative: %s }\n",
+	       bt_mesh_sensor_ch_str(&sensor->descriptor.tolerance.positive),
+	       bt_mesh_sensor_ch_str(&sensor->descriptor.tolerance.negative));
+	printk("\tsampling type: %d\n", sensor->descriptor.sampling_type);
+	printk("\tperiod: %lld\n", sensor->descriptor.period);
+	printk("\tupdate interval: %lld\n", sensor->descriptor.update_interval);
 }
 
 static const struct bt_mesh_sensor_cli_handlers bt_mesh_sensor_cli_handlers = {
 	.data = sensor_cli_data_cb,
 	.series_entry = sensor_cli_series_entry_cb,
+	.setting_status = sensor_cli_setting_status_cb,
+	.sensor = sensor_cli_desc_cb,
 };
 
-static struct bt_mesh_sensor_cli sensor_cli =
-	BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
+static struct bt_mesh_sensor_cli sensor_cli = BT_MESH_SENSOR_CLI_INIT(&bt_mesh_sensor_cli_handlers);
 
 static struct k_work_delayable get_data_work;
 
 static void get_data(struct k_work *work)
 {
-	static bool req_temp_range;
+	if (!bt_mesh_is_provisioned()) {
+		k_work_schedule(&get_data_work, K_MSEC(GET_DATA_INTERVAL));
+		return;
+	}
+
+	static uint32_t sensor_idx;
 	int err;
 
 	/* Only one message can be published at a time. Swap sensor after each timeout. */
-	if (req_temp_range) {
+	switch ((sensor_idx++ % 3)) {
+	case (0): {
+		err = bt_mesh_sensor_cli_get(&sensor_cli, NULL, &bt_mesh_sensor_present_dev_op_temp,
+					     NULL);
+		if (err) {
+			printk("Error getting chip temperature (%d)\n", err);
+		}
+		break;
+	}
+	case (1): {
 		err = bt_mesh_sensor_cli_series_entries_get(
-			&sensor_cli, NULL,
-			&bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range, NULL, NULL,
-			NULL);
+			&sensor_cli, NULL, &bt_mesh_sensor_rel_runtime_in_a_dev_op_temp_range, NULL,
+			NULL, NULL);
 		if (err) {
-			printk("Error getting relative chip temperature data (%d)\n",
-			       err);
+			printk("Error getting relative chip temperature data (%d)\n", err);
 		}
-	} else {
-		err = bt_mesh_sensor_cli_get(
-			&sensor_cli, NULL, &bt_mesh_sensor_time_since_presence_detected,
-			NULL);
+		break;
+	}
+	case (2): {
+		err = bt_mesh_sensor_cli_get(&sensor_cli, NULL,
+					     &bt_mesh_sensor_time_since_presence_detected, NULL);
 		if (err) {
-			printk("Error getting time since presence detected (%d)\n",
-			       err);
+			printk("Error getting time since presence detected (%d)\n", err);
 		}
+		break;
+	}
 	}
+	k_work_schedule(&get_data_work, K_MSEC(GET_DATA_INTERVAL));
+}
 
-	req_temp_range = !req_temp_range;
+static const struct sensor_value temp_ranges[][2] = {
+	{ { 0 }, { 100 } },
+	{ { 10 }, { 20 } },
+	{ { 22 }, { 30 } },
+	{ { 40 }, { 50 } },
+};
 
-	k_work_schedule(&get_data_work, K_MSEC(GET_DATA_INTERVAL));
+static void button_handler_cb(uint32_t pressed, uint32_t changed)
+{
+	if (!bt_mesh_is_provisioned()) {
+		return;
+	}
+
+	static uint32_t temp_idx;
+
+	if ((pressed & BIT(0))) {
+		int err;
```

Comment:
Should the err declaration be moved to the outside of the if-s?

Reply:
Yes, you can make one common instance for all buttons 👍